x-pack/plugins/alerting/server/rules_client/methods/create.ts

```diff
@@ -46,11 +46,12 @@ export interface CreateOptions<Params extends RuleTypeParams> {
     | 'nextRun'
   > & { actions: NormalizedAlertAction[] };
   options?: SavedObjectOptions;
+  skipActionConnectorsValidations?: boolean;
 }
 
 export async function create<Params extends RuleTypeParams = never>(
   context: RulesClientContext,
-  { data, options }: CreateOptions<Params>
+  { data, options, skipActionConnectorsValidations }: CreateOptions<Params>
 ): Promise<SanitizedRule<Params>> {
```

Comment:
There were some changes merged in https://github.com/elastic/kibana/pull/145637 that I believe makes this `skipActionConnectorsValidations` unnecessary. We are now checking whether the rule type to create is a SIEM rule and stripping action level frequencies. Can you verify that you shouldn't need this skip param?

cc @Zacqary

Reply:
Thanks for the explanation @WafaaNasr! Instead of bypassing the action validation altogether what do you think about a flag called `allowMissingConnectorSecrets` that is passed into the `validateActions` function. Inside that function, there is a check for connectors with missing secrets that will throw the validation error. Passing `allowMissingConnectorSecrets=true` to `validateActions` will log a warning but not throw the error? That way the rest of the action body can be correctly validated.

cc @XavierM to see if he has any suggestions for this.

Reply:
Thanks, @ymao1!  Sure I see your point!
Maybe as you mentioned we can pass the `allowMissingConnectorSecrets` to the `validateActions` function 
then add an if condition around line 31=> 50, check if  `allowMissingConnectorSecrets` was true 
to avoid getting the actions again, and loop over them to check for missing secrets.
https://github.com/elastic/kibana/blob/main/x-pack/plugins/alerting/server/rules_client/lib/validate_actions.ts#L32

Reply:
I agree with @ymao1 and also like better the name of `allowMissingConnectorSecrets`. We should log these errors because if we have SDH, it will be easy to understand why it happens and give some guidance to our users. 
 I will do something like below, really similar to what you did ;)
 ```ts
 export async function validateActions(
  context: RulesClientContext,
  alertType: UntypedNormalizedRuleType,
  data: Pick<RawRule, 'notifyWhen' | 'throttle' | 'schedule'> & {
    actions: NormalizedAlertAction[];
  },
  logger: Logger,
  allowMissingConnectorSecrets?: boolean
): Promise<void> {
  const { actions, notifyWhen, throttle } = data;
  const hasRuleLevelNotifyWhen = typeof notifyWhen !== 'undefined';
  const hasRuleLevelThrottle = Boolean(throttle);
  if (actions.length === 0) {
    return;
  }

  const errors = [];

  // check for actions using connectors with missing secrets
  const actionsClient = await context.getActionsClient();
  const actionIds = [...new Set(actions.map((action) => action.id))];
  const actionResults = (await actionsClient.getBulk(actionIds)) || [];
  const actionsUsingConnectorsWithMissingSecrets = actionResults.filter(
    (result) => result.isMissingSecrets
  );

  if (actionsUsingConnectorsWithMissingSecrets.length) {
    if (allowMissingConnectorSecrets) {
      logger.error(
        `Invalid connectors with "allowMissingConnectorSecrets": ${actionsUsingConnectorsWithMissingSecrets
          .map((connector) => connector.name)
          .join(', ')}`
      );
    } else {
      errors.push(
        i18n.translate('xpack.alerting.rulesClient.validateActions.misconfiguredConnector', {
          defaultMessage: 'Invalid connectors: {groups}',
          values: {
            groups: actionsUsingConnectorsWithMissingSecrets
              .map((connector) => connector.name)
              .join(', '),
          },
        })
      );
    }
  }
  ...
}
 ```

Reply:
Awesome, thanks @XavierM thanks for the explanation and the suggested code,
Could you please [validate](https://github.com/elastic/kibana/pull/148703/commits/31bcd730b7977074e3c52426d02433eb806b2569#diff-1864733be91bc47a8a2779c6b2eea65a06203dd355dcc6e07b815ba703af7d2eR41)